source/code/source/playfab/PlayFabCurlHttpPlugin.cpp

```diff
@@ -110,6 +112,27 @@ namespace PlayFab
         } // UNLOCK httpRequestMutex
     }
 
+    constexpr char requestIdHeaderKey[] = "X-RequestId:";
+    constexpr char whitespace[] = "\t\n\v\f\r ";
+    size_t HeaderCallback(char* buffer, size_t size, size_t nitems, void* userdata)
+    {
+        size_t headerLen = _countof(requestIdHeaderKey) - 1;
+        CallRequestContainer& reqContainer = *static_cast<CallRequestContainer*>(userdata);
+        // If this header-line is long enough, and the header starts with the key we expect
+        if ((nitems > headerLen) && (strncasecmp(buffer, requestIdHeaderKey, headerLen) == 0))
+        {
+            // The value is the requestId
+            std::string requestId = std::string(buffer + headerLen, nitems - headerLen);
+            // Trim any whitespace
+            requestId.erase(0, requestId.find_first_not_of(whitespace));
+            requestId.erase(requestId.find_last_not_of(whitespace) + 1);
```

Comment:
if the value is only whitespace, the second erase will throw out of range.

Perhaps:
```cpp
auto offset = requestId.find_first_not_of(whitespace);
if (offset != std::string::npos)
{
    requestId.erase(0, offset);
    requestId.erase(requestId.find_last_not_of(whitespace) + 1);
}
```
The second erase is now only called if the string contains non-ws